### ISSUE TITLE
Remove __reactServerBase, __setReactServerBase and __reactServerManifest from window

### ISSUE THREAD

**Issue by jhnns** [redfin/react-server#800]
Following up #794 

---

I don't think that it's necessary to put these things on the window object. Most of it could be solved via webpack or the [`chunk-manifest-webpack-plugin`](https://github.com/soundcloud/chunk-manifest-webpack-plugin) (which is currently listed in the `package.json` but not used). I don't understand why this was necessary in the first place, so maybe you could shed some light on it?

I think, removing these implicit global variables makes it easier to grasp what is actually going on in the browser. I would even argue that beside `__reactServerState` there should nothing be on the window object. But that's clearly beyond my scope.

**Comment by jhnns** [comment]
I think, we will need to put *one* manifest on the window object because this needs to be rendered into the HTML (which is not long-term cached). And if something is rendered into the HTML, it can only be accessed via `window` from the bundle.

**Comment by roblg** [comment]
Agreed, I don't think these are strictly necessary. Sounds like a good enhancement.

**Comment by doug-wade** [comment]
Disclaimer: haven't thought about this one very carefully.

Maybe we can put them into config.js, rather than attaching them to window?
